arches_controlled_lists/src/arches_controlled_lists/widgets/ReferenceSelectWidget/components/ReferenceSelectWidgetEditor.vue

```diff
@@ -1,61 +1,97 @@
     };
     nodeAlias: string;
     graphSlug: string;
 }>();
 
-const options = ref<ControlledListItem[]>(props.initialValue || []);
+const options = ref<ReferenceSelectTreeNode[]>();
 const isLoading = ref(false);
 const optionsError = ref<string | null>(null);
-
-const formFieldRef = useTemplateRef("formFieldRef");
-
-// this watcher is necessary to be able to format the value of the form field when the date picker is updated
-watch(
-    // @ts-expect-error - This is a bug in the PrimeVue types
-    () => formFieldRef.value?.field?.states?.value,
-    (newVal) => {
-        if (typeof newVal === "string") {
-            // @ts-expect-error - This is a bug in the PrimeVue types
-            formFieldRef.value!.field.states.value = [
-                options.value.find(
-                    (option: ControlledListItem) => option.uri === newVal,
-                ),
-            ];
-        }
-    },
+const expandedKeys: Ref<TreeExpandedKeys> = ref({});
+
+const initialVal = toRef(
+    extractInitialOrDefaultValue(
+        props.configuration.multiValue,
+        props.initialValue,
+        props.configuration.defaultValue,
+    ),
 );
 
+function extractInitialOrDefaultValue(
+    multiVal: boolean,
+    initialVal: ReferenceSelectFetchedOption[] | undefined,
+    defaultVal: ReferenceSelectFetchedOption[] | undefined,
+) {
+    return multiVal
+        ? initialVal
+            ? initialVal?.map((reference) => formatValForPrimevue(reference))
+            : defaultVal?.map((reference) => formatValForPrimevue(reference))
+        : initialVal
+          ? formatValForPrimevue(initialVal ? initialVal[0] : undefined)
+          : formatValForPrimevue(defaultVal ? defaultVal[0] : undefined);
+}
+
+function formatValForPrimevue(val: ReferenceSelectFetchedOption | undefined) {
+    if (!val) {
+        return undefined;
+    }
+    return { [val?.list_item_id]: true };
+}
+
+function optionAsNode(
+    item: ReferenceSelectFetchedOption,
+): ReferenceSelectTreeNode {
+    expandedKeys.value = {
+        ...expandedKeys.value,
+        [item.list_item_id]: true,
+    };
+    return {
+        key: item.list_item_id,
+        label: item.display_value,
+        children: item.children?.map(optionAsNode),
+        data: item,
+    };
+}
+
+function optionsAsNodes(
+    items: ReferenceSelectFetchedOption[],
+): ReferenceSelectTreeNode[] {
+    return items
+        .filter((item): item is ReferenceSelectFetchedOption => !!item)
+        .map(optionAsNode);
+}
+
 async function getOptions() {
     isLoading.value = true;
-
     try {
-        const fetchedLists = await fetchLists([props.nodeAlias]);
-
-        options.value = fetchedLists.controlled_lists[0].items.map(
-            (item: ControlledListItem) => ({
-                list_id: item.list_id,
-                uri: item.uri,
-                labels: item.values,
-            }),
+        const fetchedLists = await fetchWidgetOptions(
```

Comment:
this should probably exist at a higher level

Reply:
`ResourceInstanceMultiselect` widget does it's options fetching at the editor level. I've seen that it's an anti-pattern to pass functions as props in vue, so how would we keep lazy loading at the editor level but maintain this function at the parent level?